features/FEATURE_BLE/ble/common/Duration.h

```diff
@@ -231,6 +232,48 @@ struct Duration {
         return Duration(Forever::VALUE);
     }
 
+#if defined(DOXYGEN_ONLY)
+    /**
+     * Test if the forever value is being held
+     * @return True if the forever value is held False otherwise
+     */
+    bool isForever() const;
+#else
+    // Overload when Forever isn't defined
+    template<typename DefaultForever = void*>
+    std::enable_if_t<
+        std::is_same<DefaultForever, Forever>::value,
+        bool
+    >
+    isForever() const
+    {
+        return false;
+    }
+
+    // Overload when Forever is defined
+    template<typename DefaultForever = void*>
+    std::enable_if_t<
+        std::is_same<DefaultForever, Forever>::value == false,
+        bool
+    >
+    isForever() const
+    {
+        return duration == Forever::VALUE;
+    }
+#endif
+
+    /**
+     * Convert the duration into an std::chrono one.
+     * @return The duration in the std::chrono format.
+     */
+    std::chrono::duration<Rep, typename std::ratio<TB, 1000000>::type>
+    valueChrono() const
+    {
+        MBED_ASSERT(isForever() == false);
```

Comment:
Also prefer `!isForever()` - (don't feel quite as strongly as the `is_same` - doesn't take quite as long to read across to the `== false`.)

Reply:
Do not have a strong opinion either way.
[Done](https://github.com/ARMmbed/mbed-os/compare/244c974085576de2f25a69c81ea0c816bc4c22b0..a3f4cf212d6bf19f7a9885ce224fe8478b9cb57d).